torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -461,6 +461,46 @@ TensorView* abs(TensorView* tv) {
   return abs(tv->as<Val>())->as<TensorView>();
 }
 
+// The output of real(complex_tensor) are real numbers
+Val* real(Val* v) {
+  if (v->getDataType() == DataType::ComplexDouble) {
+    Val* out = newValLike(v, DataType::Double);
+    IrBuilder::create<UnaryOp>(UnaryOpType::Real, out, v);
+    return out;
+  }
+  if (v->getDataType() == DataType::ComplexFloat) {
+    Val* out = newValLike(v, DataType::Float);
+    IrBuilder::create<UnaryOp>(UnaryOpType::Real, out, v);
+    return out;
+  }
+  // We use UnaryOpType::Set instead of UnaryOpType::Real to support non-complex
+  // tensors
+  return unaryOp(UnaryOpType::Set, v);
+}
+
+TensorView* real(TensorView* tv) {
+  return real(tv->as<Val>())->as<TensorView>();
+}
+
+// The output of imag(complex_tensor) are real numbers
+Val* imag(Val* v) {
+  if (v->getDataType() == DataType::ComplexDouble) {
+    Val* out = newValLike(v, DataType::Double);
+    IrBuilder::create<UnaryOp>(UnaryOpType::Real, out, v);
+    return out;
+  }
+  if (v->getDataType() == DataType::ComplexFloat) {
+    Val* out = newValLike(v, DataType::Float);
+    IrBuilder::create<UnaryOp>(UnaryOpType::Real, out, v);
```

Comment:
We mean `Imag` instead of `Real`!?

Reply:
Can we get a cpp test that actually check the result? Just to be on the safe side.

Reply:
😱 Thank you so much for catching this! We do have a CPP test, but I have no idea why that test was not failing. I will investigate.

Reply:
:crying_cat_face:  my node timed-out right before finishing the build.... gimme a little bit more time on this one.....

Reply:
Mentioned this in the separate issue: https://github.com/csarofeen/pytorch/issues/1808#issuecomment-1176820452

Don't see a quick WAR, we can support this in integration, but since Kevin is changing that, I don't see a good point fixing it here. Let's disable the nvfuser test for now. @zasdfgbnm . cc'ing @kevinstephano

Reply:
I have updated this test to skip nvfuser tests.